test/es-module/test-esm-import-flag.js

```diff
@@ -0,0 +1,124 @@
+'use strict';
+
+require('../common');
+const fixtures = require('../common/fixtures');
+const assert = require('assert');
+const { spawnSync } = require('child_process');
+
+const cjsEntry = fixtures.path('es-modules', 'cjs-file.cjs');
+const cjsImport = fixtures.fileURL('es-modules', 'cjs-file.cjs');
+const mjsEntry = fixtures.path('es-modules', 'mjs-file.mjs');
+const mjsImport = fixtures.fileURL('es-modules', 'mjs-file.mjs');
+
+{
+  const { status, stderr, stdout } = spawnSync(
+    process.execPath,
+    [
+      '--import', mjsImport,
+      '-e', 'console.log("log")',
+    ],
+    { encoding: 'utf8' },
+  );
+
+  assert.strictEqual(status, 0);
+  assert.strictEqual(stderr, '');
+  assert.strictEqual(stdout.replace(/\r\n/g, '\n'), '.mjs file\nlog\n');
```

Comment:
If you look at the newer ESM tests, in general we prefer `assert.match` because if it fails, the non-matching text is printed as part of the error message. The use of regex also allows you to avoid dealing with cross-platform line endings for single-line matches, or to make the carriage return optional for multiline:

```suggestion
  assert.match(stdout, /\.mjs file\r?\nlog/);
```

And likewise for all the similar ones.